app/models/user.rb

```diff
@@ -20,5 +20,12 @@ class User < ApplicationRecord
   scope :enabled, -> { all }
 
   # -- Class Methods --------------------------------------------------------
+  def self.authenticate(email, pass)
+    user = find_or_create_by(email: email)
+    return user if user && BCrypt::Password.new(::Configuration.shared_password) == pass
+  end
+
```

Comment:
I'm not sure we need to add this method:

1. It seems to only be used by the Basic Auth strategy (more comments below)
2. It'd diverge from `User#authenticate` in Pro.